src/main/main.ts

```diff
@@ -34,8 +39,87 @@ export function importOnestopToBasecamp(): void {
     deleteOldRows(processedRowIds);
 }
 
+/**
+ * Processes existing event rows by checking whether the row has been modified and if so indiscriminately
+ * updates all the tasks with the new information
+ * 
+ * @param row 
+ */
 function processExistingRow(row: Row): void {
-    // Will be implemented as part of https://3.basecamp.com/4474129/buckets/38736474/todos/7717428992
+
+    Logger.log("Checking if the row has changed...\n")
+
+    if(hasChanged(row)) {
+
+        Logger.log("Changes detected in row! Updateding associated todos to reflect changes...\n")
+        
+        const savedRowBasecampMapping: RowBasecampMapping | null = getRowBasecampMapping(row);
+        
+        if(savedRowBasecampMapping == null) {
+            throw new RowBasecampMappingMissingError("The rowBasecampMapping object is null! Unable to proceed with updating the todo!");
+        }
+        
+        const basecampTodoRequests: Map<string, BasecampTodoRequest> = getBasecampTodoRequestsForRow(row);
+        const currentRoles: string[] = Array.from(basecampTodoRequests.keys());
+
+        const roleTodoIdMap = savedRowBasecampMapping.roleTodoIdMap
+        const originalRoles: string[] = Object.keys(roleTodoIdMap)
+
+        for (const role of currentRoles) {
```

Comment:
this could be broken up as well:
1. getNewRoles 
2. getExistingRoles
3. createTodosForNewRoles - needs to return todoIds to avoid side effects
4. updateTodosForExistingRoles
5. addNewRoles (to roleTodoIdMap), returns a copy with the new roles added

Reply:
createTodosForNewRoles - needs to return todoIds to avoid side effects

I can still implement this, but not sure why returning todoids avoids side effects?

Reply:
Oh wait, actually I know why now

Reply:
Okay I implemented all the new functions